### PR TITLE
Manual alignment in samplesheets

### LINKED ISSUE #38: It should be possible to provide manual image alignments on a per-sample basis
### Description of the bug

Currently, I can only specify a single "manual alignment" file for the entire workflow. However, these JSON files are inherently associated to a sample. It should be possible to provide them on a per-sample basis via the samplesheet. 

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by fasterius**
This PR adds functionality to specify manual alignment files in the samplesheets rather than via a parameter. In addition to this, the samplesheet checker has been updated to conform to the latest nf-core template, which meant major changes.

### CODE REVIEW

File: subworkflows/local/input_check.nf [hunk 1]
```diff
@@ -31,21 +31,30 @@ def create_spaceranger_channels(LinkedHashMap row) {
...
-        exit 1, "ERROR: Please check input samplesheet -> fastq_dir directory does not exist!\n${row.fastq_1}"
+        exit 1, "ERROR: Please check input samplesheet -> fastq_dir directory does not exist!\n${row.fastq_dir}"
     }
     if (!file(row.tissue_hires_image).exists()) {
-        exit 1, "ERROR: Please check input samplesheet -> tissue_hires_image file does not exist!\n${row.fastq_1}"
+        exit 1, "ERROR: Please check input samplesheet -> tissue_hires_image file does not exist!\n${row.tissue_hires_image}"
     }
-    array = [
+    if ( row.manual_alignment.isEmpty() ) {
+        manual_alignment = file ( "EMPTY_ALIGNMENT" )
```
Comment (grst) [inline subworkflows/local/input_check.nf]: It's possible to speficy an empty list (`[]`) as null token. I think it was at some point decided this is the way to go over `EMPTY_XXX`, see https://nf-co.re/docs/contributing/modules#inputoutput-options

File: subworkflows/local/input_check.nf [hunk 2]
```diff
@@ -54,29 +63,29 @@ def create_visium_channels(LinkedHashMap row) {
...
+        exit 1, "ERROR: Please check input samplesheet -> barcodes file does not exist!\n${row.barcodes}"
     }
     if (!file(row.features).exists()) {
-        exit 1, "ERROR: Please check input samplesheet -> features file does not exist!\n${row.fastq_1}"
+        exit 1, "ERROR: Please check input samplesheet -> features file does not exist!\n${row.features}"
     }
     if (!file(row.matrix).exists()) {
-        exit 1, "ERROR: Please check input samplesheet -> matrix file does not exist!\n${row.fastq_1}"
+        exit 1, "ERROR: Please check input samplesheet -> matrix file does not exist!\n${row.matrix}"
     }
```
Comment (grst) [inline subworkflows/local/input_check.nf:87]: consider using a loop?

**Review by grst — APPROVED** [review]

File: subworkflows/local/input_check.nf (hunk 1 above)
Comment (fasterius) [reply subworkflows/local/input_check.nf]: Thanks, I wasn't aware of this new practice!

File: modules/local/spaceranger_count.nf [hunk 3]
```diff
@@ -28,8 +27,8 @@ process SPACERANGER_COUNT {
     path "versions.yml"                                      , emit: versions
 
     script:
-    def probeset         = probeset.name != 'EMPTY_PROBESET' ? "--probe-set=${probeset}" : ''
-    def manual_alignment = manual_alignment.name != 'EMPTY_ALIGNMENT' ? "--loupe-alignment=${manual_alignment}" : ''
+    def probeset         = probeset.name != [] ? "--probe-set=${probeset}" : ''
```
Comment (grst) [inline modules/local/spaceranger_count.nf]: That's actually the beauty of using `[]`. Groovy evaluates this as false, so you can just do

```
probeset ? "--probe-set=${probeset}" : ""
```

But this part is anyway going to be replaced with the spaceranger from `modules`...

Comment (fasterius) [reply modules/local/spaceranger_count.nf]: Oh, that's quite a bit less verbose! Yeah it'll go away soon, but I'll change it in the meantime anyway.

**Pull request merged**